mlflow/store/artifact/uc_volumes_artifact_repo.py

```diff
@@ -0,0 +1,217 @@
+import json
+import os
+import posixpath
+
+import mlflow.utils.databricks_utils
+from mlflow.entities import FileInfo
+from mlflow.environment_variables import MLFLOW_ENABLE_DBFS_FUSE_ARTIFACT_REPO
+from mlflow.exceptions import MlflowException
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.store.artifact.local_artifact_repo import LocalArtifactRepository
+from mlflow.store.tracking.rest_store import RestStore
+from mlflow.tracking._tracking_service import utils
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.file_utils import relative_path_to_artifact_path
+from mlflow.utils.rest_utils import RESOURCE_NON_EXISTENT, http_request, http_request_safe
+from mlflow.utils.string_utils import strip_prefix
+from mlflow.utils.uri import (
+    get_databricks_profile_uri_from_artifact_uri,
+    is_databricks_model_registry_artifacts_uri,
+    is_valid_dbfs_uri,
+    remove_databricks_profile_info_from_artifact_uri,
+)
+
+# The following constants are defined as @developer_stable
+DIR_API_ENDPOINT = "/api/2.0/fs/directories"
+FILE_API_ENDPOINT = "/api/2.0/fs/files"
+DOWNLOAD_CHUNK_SIZE = 1024
+
+
+class UCVolumesRestArtifactRepository(ArtifactRepository):
+    """
+    Stores artifacts on Volumes using the Files REST API.
+
+    This repository is used with URIs of the form `dbfs:/Volumes/<path>`. The repository can only be
+    used together with the RestStore.
+    """
+
+    def __init__(self, artifact_uri):
+        # TODO: add is_valid_volumes_uri
+        if not is_valid_dbfs_uri(artifact_uri):
+            raise MlflowException(
+                message="DBFS URI must be of the form dbfs:/Volumes/<path>",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+
+        # The dbfs:/ path ultimately used for artifact operations should not contain the
+        # Databricks profile info, so strip it before setting ``artifact_uri``.
+        super().__init__(remove_databricks_profile_info_from_artifact_uri(artifact_uri))
+
+        databricks_profile_uri = get_databricks_profile_uri_from_artifact_uri(artifact_uri)
+        if databricks_profile_uri:
+            hostcreds_from_uri = get_databricks_host_creds(databricks_profile_uri)
+            self.get_host_creds = lambda: hostcreds_from_uri
+        else:
+            self.get_host_creds = _get_host_creds_from_default_store()
+
+    def _databricks_api_request(self, endpoint, method, **kwargs):
+        host_creds = self.get_host_creds()
+        return http_request_safe(host_creds=host_creds, endpoint=endpoint, method=method, **kwargs)
+
+    def _volumes_list_api(self, path):
+        host_creds = self.get_host_creds()
+        return http_request(
+            host_creds=host_creds, endpoint=f"{DIR_API_ENDPOINT}{path}", method="GET"
+        )
+
+    def _volumes_download(self, output_path, endpoint):
```

Comment:
can this be changed to a verb-noun pattern in keeping with method naming conventions?